dialogue-core/src/main/java/com/palantir/dialogue/core/DeprecationWarningChannel.java

```diff
@@ -101,9 +105,11 @@ static EndpointChannel create(Config cf, EndpointChannel delegate, Endpoint endp
      * penalty for failing to rate limit correctly is a few extra log lines, and we choose that penalty over the cost
      * of synchronization.
      */
-    private boolean tryAcquire(String serviceName) {
-        if (loggingRateLimiter.getIfPresent(serviceName) == null) {
-            loggingRateLimiter.put(serviceName, SENTINEL);
+    private boolean tryAcquire(String channelName, Endpoint endpoint) {
+        LoggingRateLimiterKey key =
+                LoggingRateLimiterKey.of(channelName, endpoint.serviceName(), endpoint.endpointName());
```

Comment:
we could create this key upfront to save recreating it on every RPC? (like ~L76?)

Reply:
yes and no -- I considered it but I think it would be harmful for the jaxrs shim. I don't want to optimize for the shim, but I don't expect most calls to hit deprecated endpoints.